OBAKitCore/Services/Persistence/PersistenceService.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  PersistenceService.swift
+//  OBAKitCore
+// 
+//  Copyright © 2023 Open Transit Software Foundation.
+//  This source code is licensed under the Apache 2.0 license found in the
+//  LICENSE file in the root directory of this source tree.
+//
+
+import Foundation
+import os.log
+import GRDB
+
+public protocol ReferencesProvider {
+    var references: References? { get }
+}
+
+extension RESTAPIResponse: ReferencesProvider { }
+
+public actor PersistenceService {
+    public struct Configuration {
+        public enum DatabaseLocation {
+            case disk
+            case memory
+        }
+
+        public let regionIdentifier: Int
+        public let databaseLocation: DatabaseLocation
+
+        public let tableCreators: [DatabaseTableCreator.Type] = [
+            Agency.self,
+            Stop.self,
+            Route.self,
+            Trip.self,
+            TripDetails.self,
+            Situation.self
+        ]
+
+        public init(regionIdentifier: Int, databaseLocation: DatabaseLocation) {
+            self.regionIdentifier = regionIdentifier
+            self.databaseLocation = databaseLocation
+        }
+    }
+
+    public let configuration: Configuration
+    public let logger = os.Logger(subsystem: "org.onebusaway.iphone", category: "PersistenceService")
+    public let database: DatabaseQueue
+
+    public init(_ configuration: Configuration) throws {
+        self.configuration = configuration
+
+        switch configuration.databaseLocation {
+        case .disk:
+            let fileManager = FileManager.default
+            let applicationSupportURL = try fileManager.url(for: .applicationSupportDirectory, in: .userDomainMask, appropriateFor: nil, create: true)
+            let directoryURL = applicationSupportURL
+                .appendingPathComponent("regions", isDirectory: true)
+                .appendingPathComponent("\(configuration.regionIdentifier)", isDirectory: true)
+            try fileManager.createDirectory(at: directoryURL, withIntermediateDirectories: true)
+
+            let databaseURL = directoryURL.appendingPathComponent("onebusaway.sqlite")
+
+            logger.info("Opening SQLite at \(databaseURL, privacy: .public).")
+            database = try DatabaseQueue(path: databaseURL.path)
+        case .memory:
+            logger.info("Opening SQLite in memory.")
+            database = try DatabaseQueue()
+        }
+
+        // Migration registration
+        var migrator = DatabaseMigrator()
+        migrator.registerMigration("createUsingTableCreators") { db in
+            let logger = os.Logger(subsystem: "org.onebusaway.iphone", category: "PersistenceService-DatabaseMigrator")
```

Comment:
`subsystem`'s value probably should not be hardcoded here.

Reply:
Changed to whitelabel-friendly `OBAKitCore`.